pdns/dnsdist-lua.hh

```diff
@@ -107,3 +107,37 @@ void setupLuaInspection(LuaContext& luaCtx);
 void setupLuaVars(LuaContext& luaCtx);
 void setupLuaWeb(LuaContext& luaCtx);
 void setupLuaLoadBalancingContext(LuaContext& luaCtx);
+
+/**
+ * getOptionalValue(vars, key, value)
+ *
+ * Attempts to extract value for key in vars.
+ * Erases the key from vars.
+ *
+ * returns: -1 if type wasn't compatible, 0 if not found or number of element(s) found
+ */
+template<class G, class T, class V>
+static inline int getOptionalValue(boost::optional<V>& vars, const std::string& key, T& value) {
+  /* nothing found, nothing to return */
+  if (!vars)
+    return 0;
+  if (vars->count(key)) {
+    try {
+      value = boost::get<G>((*vars)[key]);
+    } catch (boost::bad_get& e) {
+      /* key is there but isn't compatible */
+      return -1;
```

Comment:
Would it make sense to log a warning there? I don't think we used to but it might be a nice improvement?

Reply:
This is there mostly to facilitate
``` 

      if (getOptionalValue<std::string>(vars, "pool", valueStr) > 0) {
        ret->pools.insert(valueStr);
      } else if (getOptionalValue<decltype(pools)>(vars, "pool", pools) > 0) {
        for(auto& p : pools) {
          ret->pools.insert(p.second);
        }
      }
 ```

Reply:
Right, I thought we could catch an invalid value that way, but I guess it's not practical.